### ISSUE TITLE
Adding auth module with custom configuration creates roles with undefined names.

### ISSUE THREAD

**Issue by akefirad** [aws-amplify/amplify-cli#4930]
**Describe the bug**
Trying to add auth module with custom configuration creates two roles with `undefined` in the name:
```
RoleName: !If [ShouldNotCreateEnvResources, 'testam52d79772_sns-role', !Join ['',[ 'sns', 'undefined', !Select [3, !Split ['-', !Ref 'AWS::StackName']], '-', !Ref env]]]
...
RoleName: !If [ShouldNotCreateEnvResources, !Ref userpoolClientLambdaRole, !Join ['',['upClientLambdaRole', 'undefined', !Select [3, !Split ['-', !Ref 'AWS::StackName']], '-', !Ref env]]]

```

**Amplify CLI Version**
4.24.3

**To Reproduce**
```
# amplify add auth
Using service: Cognito, provided by: awscloudformation

 The current configured provider is Amazon Cognito.

 Do you want to use the default authentication and security configuration? Manual configuration
 Select the authentication/authorization services that you want to use: User Sign-Up & Sign-In only (Best used with a cloud API only)
 Please provide a friendly name for your resource that will be used to label this category in the project: testamplify52d7977252d79772
 Please provide a name for your user pool: testamplify52d79772_userpool_52d79772
 Warning: you will not be able to edit these selections.
 How do you want users to be able to sign in? Username
 Do you want to add User Pool Groups? No
 Do you want to add an admin queries API? Yes
? Do you want to restrict access to the admin queries API to a specific Group No
 Multifactor authentication (MFA) user login options: OPTIONAL (Individual users can use MFA)
 For user login, select the MFA types: SMS Text Message, Time-Based One-Time Password (TOTP)
 Please specify an SMS authentication message: Your authentication code is {####}
 Email based user registration/forgot password: Enabled (Requires per-user email entry at registration)
 Please specify an email verification subject: Your verification code
 Please specify an email verification message: Your verification code is {####}
 Do you want to override the default password policy for this User Pool? No
 Warning: you will not be able to edit these selections.
 What attributes are required for signing up? Email
 Specify the app's refresh token expiration period (in days): 30
 Do you want to specify the user attributes this app can read and write? No
 Do you want to enable any of the following capabilities?
 Do you want to use an OAuth flow? No
? Do you want to configure Lambda Triggers for Cognito? No
Successfully added resource testamplify52d7977252d79772 locally
```

**Expected behavior**
```
RoleName: !If [ShouldNotCreateEnvResources, 'testam955f59ac_sns-role', !Join ['',[ 'sns', '955f59ac', !Select [3, !Split ['-', !Ref 'AWS::StackName']], '-', !Ref env]]]
...
RoleName: !If [ShouldNotCreateEnvResources, !Ref userpoolClientLambdaRole, !Join ['',['upClientLambdaRole', '955f59ac', !Select [3, !Split ['-', !Ref 'AWS::StackName']], '-', !Ref env]]]
```

**Desktop (please complete the following information):**
```
# sw_vers
ProductName:	Mac OS X
ProductVersion:	10.15.5
BuildVersion:	19F101
# node -v
v14.5.0
```

**Comment by akshbhu** [comment]
Hi @akefirad 

This issue has been fixed. You can try with the latest version.

Let me know if this works for you?

**Comment by akefirad** [comment]
As far as I can see, no, the `undefined` is still there:
```diff
-      RoleName: !If [ShouldNotCreateEnvResources, 'wefarb1d2d5f67_sns-role', !Join ['',[ 'sns', '1d2d5f67', !Select [3, !Split ['-', !Ref 'AWS::StackName']], '-', !Ref env]]]
-      AssumeRolePolicyDocument:
+      RoleName: !If [ShouldNotCreateEnvResources, 'wefarb1d2d5f67_sns-role', !Join ['',[ 'sns', 'undefined', !Select [3, !Split ['-', !Ref 'AWS::StackName']], '-', !Ref env]]]
+      AssumeRolePolicyDocument:
```
Let me know what information you need to troubleshoot this.

**Comment by akefirad** [comment]
Just to be clear, I manually changed the names (replaced `undefined` with the ID). That's why you see it's replacing the ID with a `undefined`.

**Comment by akefirad** [comment]
Is there any update on this?

**Comment by akshbhu** [comment]
Hi @akefirad 

Just to reproduce , when you did ```amplify update auth```, which of the following you selected

```
❯ Apply default configuration with Social Provider (Federation) 
  Walkthrough all the auth configurations 
  Create or update Cognito user pool groups 
  Create or update Admin queries API 
```

For your context ,we have refractored the auth category which has fixed this ```undefined``` bug, (though it will not affect any  working flow but still doesnt looks good) as previously this was occuring in auth manual configuration. The problem was we have an ```sharedID``` which we used to define sns role which was not being populated before in manual auth configuration.

**Comment by akefirad** [comment]
`Walkthrough all the auth configurations`

**Comment by akshbhu** [comment]
Hi @akefirad 

Closing this as it  has been resolved in our latest version(4.41.2). Feel free to comment if this issue still persists

**Comment by akefirad** [comment]
I'm not sure if the fix is tested against existing stacks. Because it still generates names with `undefined` as show above.
